coalib/settings/SectionFilling.py

```diff
@@ -45,7 +45,10 @@ def fill_section(section, acquire_settings, log_printer, bears):
 
     # Get missing ones.
     if len(needed_settings) > 0:
-        new_vals = acquire_settings(None, needed_settings, section)
+        new_vals = (acquire_settings(None, needed_settings)
+                    if acquire_settings.__defaults__
+                    else
+                    acquire_settings(None, needed_settings, section))
```

Comment:
> Only the fail_acquire_settings function has a default parameter None. Rest acquire_settings function dont have default parameters so if the tuple acquire_settings.defaults is not empty , fail_acquire_settings will be called else acquire_settings will be called ;)

Actually I don't understand that. If the defaults-dict is empty, we would rather want to use the "new" calling-convention without `section`, right? Now it's the opposite.
However, really check for the number of parameters and not for defaults, as this disables any new function which might use a default for a different parameter.

Reply:
Oh and btw: We need a deprecation message again :+1:

Reply:
@Makman2 but both the acquire_settings and fail_acquire_settings functions have 3 parameters. We cant differentiate on the basis of no. of parameters but only the fail_acquire settings has a default parameter section=None . If the default parameter tuple is not empty then we can safely state that fail_acquire_settings is the one we are looking for

Reply:
The only way round is to remove the section parameter completely from Interactions.fail_acquire_settings

Reply:
#4907